manuscripts2/new_functions.py

```diff
@@ -0,0 +1,510 @@
+import logging
+
+from collections import OrderedDict, defaultdict
+from datetime import timezone
+from dateutil import parser
+
+import pandas as pd
+from elasticsearch import Elasticsearch
+from elasticsearch_dsl import A, Q, Search
+
+logger = logging.getLogger(__name__)
+
+
+class Index():
+    """Index class which will represent an index in elasticsearch"""
+
+    def __init__(self, index, url=None):
+        """
+        :url: the address/url of the elasticsearch instance to be used?
+              default: http://localhost:9200/ (optional)
+        :index: name of the elasticsearch index that is to be queried (required)
+        """
+
+        self.url = url if url else None
+        self.index = index  # index to query data from
+        if self.url:
+            self.es = Elasticsearch(self.url)  # This allows us to query _any_ elasticsearch instance
+        else:
+            logger.debug("No custom url provided, using default: http://localhost:9200/")
+            self.es = Elasticsearch("http://localhost:9200/")
+
+
+class Query():
+    """
+    The base class of all the necessary metrics that are going to be calculated.
+    """
+
+    def __init__(self, index_obj, esfilters={}, interval=None, offset=None):
```

Comment:
Since everything is an object, usually it is better to use names such as `index` instead of `index_obj`. The description of the parameters, below in the docstring, should specify which type `index` should be.